src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,34 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+    Animal.alive.push(this);
+  }
 }
+Animal.alive = [];
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health, hidden = false) {
+    super(name, health);
+    this.hidden = hidden;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  bite(prey) {
+    if (prey.hasOwnProperty('hidden') && !prey.hidden) {
+      prey.health -= 50;
+    }
+
+    if (prey.health === 0) {
+      Animal.alive.splice(Animal.alive.indexOf(prey), 1);
```

Comment:
let's make this logic a static method of Animal class. and call it here
it could be something like

```javascript
Animal.removeDead = (...
```